### ISSUE TITLE
context.report fails to parse opts into message

### ISSUE THREAD

**Issue by iancmyers** [eslint/eslint#642]
When using all of `context.report`'s options, the message will not be parsed with opts:

``` js
context.report(node, node.loc.end, "Problem with {{type}}", {type: node.type});

// Expected: "Problem with ArrayExpression"
// Actual: "Problem with {{type}}"
```
